pkg/agent/client/client.go

```diff
@@ -465,3 +465,15 @@ func (c *client) newAgentClient(ctx context.Context) (agentv1.AgentClient, *node
 	c.connections.AddRef()
 	return c.createNewAgentClient(c.connections.conn), c.connections, nil
 }
+
+// addErrorFields add fields of gRPC call status in logger
+func (c *client) addErrorFields(err error) logrus.FieldLogger {
+	if s, ok := status.FromError(err); ok {
+		c.c.Log = c.c.Log.WithFields(logrus.Fields{
+			telemetry.StatusCode:    s.Code(),
+			telemetry.StatusMessage: s.Message(),
+		})
+	}
+
+	return c.c.Log.WithError(err)
+}
```

Comment:
```suggestion
func (c *client) withErrorFields(err error) logrus.FieldLogger {
	if err == nil {
		return c.c.Log
	}
	
	logger := c.c.Log.WithError(err)
	if s, ok := status.FromError(err); ok {
		logger = c.c.Log.WithFields(logrus.Fields{
			telemetry.StatusCode:    s.Code(),
			telemetry.StatusMessage: s.Message(),
		})
	}

	return logger
}
```
We shouldn't modify the logger in the `client` struct since the added fields will continue to be emitted in subsequent log messages. We only want these fields to be emitted in the next log message, so we need to return a new `logrus.FieldLogger` reference.

A couple other suggestions:
- If we check if `err` is `nil`, then we can just call this method after every SPIRE RPC rather than having to call it for each possible error log statement, since in some cases we log different messages based on the error.
- `withErrorFields()` is a more consistent name with Logrus's API design